File: src/protocol/BlindsUPOnTimeBuilder.cpp
#include "BlindsUPOnTimeBuilder.h"
#include "BlindsUPOnTimeCommand.h"

#include <BytesConverter.h>

using namespace std;
using namespace utility;
using namespace communication;

BlindsUPOnTimeBuilder::BlindsUPOnTimeBuilder()
{}

BlindsUPOnTimeBuilder::~BlindsUPOnTimeBuilder()
{}

unique_ptr<Command> BlindsUPOnTimeBuilder::create(const vector<uint8_t> &commandInBytes)
{
    uint32_t epochTime;

    epochTime = BytesConverter::fromVectorOfUINT8toUINT32(commandInBytes, INIT_EPOCH_TIME_POSITION_IN_FRAME);

    auto command = make_unique<BlindsUPOnTimeCommand>(epochTime);

    return move(command);
}

File: mobile_client/app/src/main/java/protocol/BlindsUPOnTimeBuilder.java
package protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class BlindsUPOnTimeBuilder implements CommandBuilder, Closeable
	{
		public BlindsUPOnTimeBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			int dateAndTime = BytesConverter.fromVectorOfUINT8toUINT32(commandInBytes, INIT_EPOCH_TIME_POSITION_IN_FRAME);
			Command command  = new BlindsUPOnTimeCommand(dateAndTime);
			return command;
		}

		private final int INIT_EPOCH_TIME_POSITION_IN_FRAME = 6;
	}